### PR TITLE
Update Raspberry Pi dev env setup to be in sync with toolchain updates

### PR DESCRIPTION
**Pull request by nicolaerosia**
Devguide update for https://github.com/PX4/Firmware/pull/7248
Signed-off-by: Nicolae Rosia <nicolae.rosia@gmail.com>

### CODE REVIEW

**Review by LorenzMeier — APPROVED** [review]

**Comment by LorenzMeier** [comment]
@bkueng Can you review as well and merge if ok?

File: en/setup/dev_env_linux.md [hunk 1]
```diff
@@ -128,21 +128,45 @@ Note: Alternatively, especially on Mac, you can also use [nano-dm](https://githu
...
+Developers working on Raspberry Pi hardware need to download a ARMv7 cross-compiler, either GCC or clang.
+The recommended toolchain for raspbian is GCC 4.8.3 and can be cloned from `https://github.com/raspberrypi/tools.git`.
+The `PATH` environmental variable should include the path to the gcc cross-compiler collection of tools (e.g. gcc, g++, strip) prefixed with `arm-linux-gnueabihf-`.
 
 ```sh
-git clone https://github.com/pixhawk/rpi_toolchain.git
-cd rpi_toolchain
-./install_cross.sh
-```
+git clone https://github.com/raspberrypi/tools.git ${HOME}/aero-workspace/rpi-tools
```
Comment (bkueng) [inline en/setup/dev_env_linux.md]: why `aero-workspace`?

File: en/setup/dev_env_linux.md [hunk 2]
```diff
@@ -128,21 +128,45 @@ Note: Alternatively, especially on Mac, you can also use [nano-dm](https://githu
...
 
-You will be required to enter your password for toolchain installation to complete successfully.
+# test compiler
+$HOME/aero-workspace/rpi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin/arm-linux-gnueabihf-gcc -v
 
-You can pass a different path to the installer script if you wouldn't like to install the toolchain to the default location of `/opt/rpi_toolchain`. Run `./install_cross.sh <PATH>`. The installer will automatically configure required environment variables as well.
+# permanently update PATH variable by modifying ~/.profile
+echo 'export PATH=$PATH:$HOME/aero-workspace/rpi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin/arm-linux-gnueabihf-' >> ~/.profile
```
Comment (bkueng) [inline en/setup/dev_env_linux.md]: Shouldn't this be `echo 'export PATH=$PATH:$HOME/aero-workspace/rpi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin' >> ~/.profile`?

File: en/setup/dev_env_linux.md [hunk 3]
```diff
@@ -128,21 +128,45 @@ Note: Alternatively, especially on Mac, you can also use [nano-dm](https://githu
...
+Download clang for your specific distribution from [LLVM Download page](http://releases.llvm.org/download.html) and unpack it.
+Assuming that you've unpacked clang to `CLANG_DIR`, and `clang` binary is available in `CLANG_DIR/bin`, and you have the GCC cross-compiler in `GCC_DIR`, you will need to setup the symlinks for clang in the `GCC_DIR` bin dir, and add `GCC_DIR/bin` to `PATH`.
+
+Example below for building PX4 firmware out of tree, using CMake.
+```sh
+ln -s <CLANG_DIR>/bin/clang <GCC_DIR>/bin/clang
+ln -s <CLANG_DIR>/bin/clang++ <GCC_DIR>/bin/clang++
+export PATH=<GCC_DIR>/bin:$PATH
+
+mkdir /tmp/px4-build
```
Comment (bkueng) [inline en/setup/dev_env_linux.md]: I think I'd prefer something like:
```
cd <PATH-TO-PX4-SRC>
mkdir build_posix_rpi_cross_clang
cd build_posix_rpi_cross_clang
cmake <options...> ..
```
Just to be consistent with how we currently build things.

**Review by bkueng — COMMENTED** [review]
Thanks for the update. I added some comments, see below.

File: en/setup/dev_env_linux.md (hunk 1 above)
Comment (nicolaerosia) [reply en/setup/dev_env_linux.md]: no reason, I can change that

File: en/setup/dev_env_linux.md (hunk 2 above)
Comment (nicolaerosia) [reply en/setup/dev_env_linux.md]: You are right

File: en/setup/dev_env_linux.md (hunk 3 above)
Comment (nicolaerosia) [reply en/setup/dev_env_linux.md]: Ok, sure

**Comment by nicolaerosia** [comment]
@bkueng I have updated to address your comments

**Pull request merged**